src/modules/dart.rs

```diff
@@ -0,0 +1,80 @@
+use std::process::Command;
+
+use super::{Context, Module, RootModuleConfig, SegmentConfig};
+
+use crate::configs::dart::DartConfig;
+
+/// Creates a module with the current Dart version
+///
+/// Will display the Dart version if any of the following criteria are met:
+///     - Current directory contains a `.dart` file
+///     - Current directory contains a `pubspec.yaml` file
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let is_dart_project = context
+        .try_begin_scan()?
+        .set_files(&["pubspec.yaml"])
+        .set_extensions(&["dart"])
+        .is_match();
+
+    if !is_dart_project {
+        return None;
+    }
+
+    match get_dart_version() {
```

Comment:
the match is not needed here, you can just use `?` to propagate the errors instead of using match
```rust
let formatted_version = format_dart_version(&get_dart_version()?)?;

    let mut module = context.new_module("dartlang");
    let config: DartConfig = DartConfig::try_load(module.config);

    module.set_style(config.style);
    module.create_segment("symbol", &config.symbol);

    module.create_segment("version", &config.version.with_value(&formatted_version));
```
    Some(module)
```

Reply:
Thanks for the tip! Will refactor that.